packages/pigeon/platform_tests/test_plugin/example/ios/RunnerTests/InstanceManagerTests.swift

```diff
@@ -0,0 +1,144 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import XCTest
+
+@testable import test_plugin
+
+final class InstanceManagerTests: XCTestCase {
+  func testAddDartCreatedInstance() {
+    let finalizerDelegate = EmptyFinalizerDelegate()
+    let instanceManager = ProxyApiTestsPigeonInstanceManager(finalizerDelegate: finalizerDelegate)
+    let object = NSObject()
+
+    instanceManager.addDartCreatedInstance(object, withIdentifier: 0)
+    XCTAssertEqual(instanceManager.instance(forIdentifier: 0), object)
+    XCTAssertEqual(instanceManager.identifierWithStrongReference(forInstance: object), 0)
+  }
+
+  func testAddHostCreatedInstance() {
+    let finalizerDelegate = EmptyFinalizerDelegate()
+    let instanceManager = ProxyApiTestsPigeonInstanceManager(finalizerDelegate: finalizerDelegate)
+    let object = NSObject()
+    _ = instanceManager.addHostCreatedInstance(object)
+
+    let identifier = instanceManager.identifierWithStrongReference(forInstance: object)
+    XCTAssertNotNil(identifier)
```

Comment:
optional nit: can use `XCTUnwrap` to avoid `!`. When it's nil, instead of crashing, it would mark it as a test failure and continue with other tests.

Reply:
I changed the following line to: 

```swift
XCTAssertEqual(instanceManager.instance(forIdentifier: try XCTUnwrap(identifier)), object)
```

Is that what you were thinking? The `XCTAssertNotNil(identifier)` check also fails the test but continues to run other tests.